jwst/ami/mask_definition_ami.py

```diff
@@ -0,0 +1,124 @@
+#
+#  Module for defining mask geometry in pupil space
+#  Replace old mask_definitions by reading info in nrm_model into a similar object.
+#
+
+import numpy as np
+import math
+
+from .utils import rotate2dccw
+
+m = 1.0
+mm = 1.0e-3 * m
+um = 1.0e-6 * m
+
+
+class NRM_definition():
+
+	def __init__(self, nrm_model, maskname='jwst_ami', rotdeg=None, chooseholes=None):
+		"""
+		Set attributes of NRM_definition class.
+
+		Parameters
+		----------
+		nrm_model: NRMModel
+			datamodel containing NRM reference file data
+		maskname: string
+			Identifier for mask geometry; default 'jwst_ami', optional
+		rotdeg: list of floats
+			range of rotations to search (degrees), optional
+		chooseholes: list
+			None, or e.g. ['B2', 'B4', 'B5', 'B6'] for a four-hole mask, optional
+		"""
+
+		if maskname not in ['jwst_ami','jwst_g7s6c']:
+			raise ValueError("Mask name not supported")
+
+		self.maskname = maskname # there's only one mask but this is used in oifits
+
+		self.read_nrm_model(nrm_model, chooseholes=chooseholes)
+
+		if rotdeg is not None:
+			self.rotdeg = rotdeg
+
+	def read_nrm_model(self, nrm_model, chooseholes=None):
+		"""
+
+		Calculate hole centers with appropriate rotation,
+		set these and other mask geometry details from NRMModel as
+		attributes.
+
+		Parameters
+		----------
+		nrm_model: NRMModel
+			datamodel containing NRM reference file data 
+		chooseholes: list
+			None, or e.g. ['B2', 'B4', 'B5', 'B6'] for a four-hole mask
+
+		Returns
+		-------
+		f2f: float
+			flat-to-flat distance of mask holes
+		ctrs_asbuilt: array
+			Actual hole centers [meters]
+		"""
+
+		ctrs_asdesigned = np.array([[nrm_model.x_a1, nrm_model.y_a1],        # B4 -> B4
+							[nrm_model.x_a2, nrm_model.y_a2],       # C5 -> C2
+							[nrm_model.x_a3, nrm_model.y_a3],       # B3 -> B5
+							[nrm_model.x_a4, nrm_model.y_a4],       # B6 -> B2
+							[nrm_model.x_a5, nrm_model.y_a5],       # C6 -> C1
+							[nrm_model.x_a6, nrm_model.y_a6],        # B2 -> B6
+							[nrm_model.x_a7, nrm_model.y_a7]])        # C1 -> C6
+
+		self.hdia = nrm_model.flat_to_flat
+		self.activeD = nrm_model.diameter
+		self.OD = nrm_model.pupil_circumscribed
```

Comment:
Any attributes that are assigned to inside these functions should be initialized in the init function.  You can just set them to None if there's no sensible default value.

Reply:
Ah, never mind, I see you addressed this in another comment!